### PR TITLE
Topic recognition : knee-classification

### PR DESCRIPTION
**Pull request by DH-Stack**
**Author**
Donghong Zhuang (44994765)

**Problem**
 Classify laterality (left or right-sided knee) of the OAI AKOA knee data set having a minimum accuracy
of 0.9 on the test set. 

**Description**

In this project, classify (left or right-sided knee) of the [OAI AKOA](https://nda.nih.gov/oai/) knee data set.

- Used Transfer learning techniques with [inceptionV3](https://en.wikipedia.org/wiki/Inceptionv3)

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Code appear valid for Easy Difficulty
- Results are good
- Commit messages reasonably detailed
- Model uses InceptionV3 but there has been enough customisation to justify (transfer learning)

File: recognition/knee-classification/model.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import tensorflow as tf
```
Comment (shakes76) [inline recognition/knee-classification/model.py:1]: Remember to add header blocks for each file in your code

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
Header block would be good
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme sparse
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Good Request Description. Well done.

**Comment by rexsw** [comment]
Code appear valid for Easy Difficulty
Results are good and well communicated 

Comments are not useful in many places for example:
    # Get model summary details
    def get_model_summary(self, model_type):  
Such comments add nothing to understanding the code 

Better places to comment could be places like
        inputs = tf.keras.Input(shape=self.img_shape)
        x = data_augmentation(inputs)
        x = preprocess_input(x)
A quick comment explaining the type of data augmentation and preprocess used save the reader from having to go find those functions
